File: app/src/main/java/com/example/firabasefirstexperience/MainActivity.java
package com.example.firabasefirstexperience;

import android.content.Intent;
import android.provider.ContactsContract;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.firabasefirstexperience.Fragments.ChatsFragment;
import com.example.firabasefirstexperience.Fragments.UserFragment;
import com.example.firabasefirstexperience.Profile.User;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;

import de.hdodenhof.circleimageview.CircleImageView;

public class MainActivity extends AppCompatActivity {

    private CircleImageView profileImage;
    private TextView tv_UserName;
    private FirebaseUser firebaseUser;
    private DatabaseReference reference;
    private FirebaseDatabase database;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolBar);
        setSupportActionBar(toolbar);
        toolbar.inflateMenu(R.menu.menu_main);

        profileImage = (CircleImageView) findViewById(R.id.profileImage);
        tv_UserName = (TextView) findViewById(R.id.tv_UserName);

        firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
        reference = FirebaseDatabase.getInstance().getReference("Users").child(firebaseUser.getUid());



        reference.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                User user = dataSnapshot.getValue(User.class);
                tv_UserName.setText(user.getUserName());

                if(user.getImageURL() != null) {
                    Glide.with(MainActivity.this).load(user.getImageURL()).into(profileImage);
                }
            }
            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {
            }
        });




        TabLayout tabLayout_MainActivity = findViewById(R.id.tabLayout_MainActivity);
        ViewPager vP_MainActivity = findViewById(R.id.vP_MainActivity);


        PageAdapter pageAdapter = new PageAdapter(getSupportFragmentManager());
        pageAdapter.addFragment(new ChatsFragment(), "Chats");
        pageAdapter.addFragment(new UserFragment(),"Users");

        vP_MainActivity.setAdapter(pageAdapter);
        tabLayout_MainActivity.setupWithViewPager(vP_MainActivity);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        switch (item.getItemId()) {
            case R.id.item_logout:
                FirebaseAuth.getInstance().signOut();
                startActivity(new Intent(MainActivity.this , StartActivity.class));
                finish();


                break;
        }
        return super.onOptionsItemSelected(item);
    }


    class PageAdapter extends FragmentPagerAdapter {

        ArrayList<Fragment> fragments;
        ArrayList<String> titles;

        public PageAdapter(FragmentManager fm) {
            super(fm);
            this.fragments = new ArrayList<>();
            this.titles = new ArrayList<>();

        }

        @Override
        public Fragment getItem(int i) {
            return fragments.get(i);
        }

        @Override
        public int getCount() {
            return fragments.size();
        }

        public void addFragment (Fragment fragment , String title) {
            fragments.add(fragment);
            titles.add(title);
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return titles.get(position);
        }
    }

}

File: app/src/main/java/com/nassdk/supchat/presentation/main/ui/MainActivity.kt
package com.nassdk.supchat.presentation.main.ui

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.bumptech.glide.Glide
import com.nassdk.supchat.R
import com.nassdk.supchat.presentation.main.mvp.MainActivityPresenter
import com.nassdk.supchat.presentation.main.mvp.MainActivityView
import com.nassdk.supchat.model.User
import com.nassdk.supchat.presentation.profilescreen.ui.ProfileActivity
import com.nassdk.supchat.presentation.startscreen.ui.StartActivity
import com.google.android.material.navigation.NavigationView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.database.*
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.navigation_header.*

class MainActivity : MvpAppCompatActivity(), MainActivityView, NavigationView.OnNavigationItemSelectedListener {


    private lateinit var fbUser: FirebaseUser
    private lateinit var reference: DatabaseReference
    private lateinit var drawerLayout: DrawerLayout

    @InjectPresenter
    lateinit var mainPresenter: MainActivityPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
    }

    private fun initViews() {
        fbUser = FirebaseAuth.getInstance().currentUser!!
        reference = FirebaseDatabase.getInstance().getReference("Users").child(fbUser.uid)

        setSupportActionBar(toolBarChats)
        toolBarChats?.inflateMenu(R.menu.menu_main)


        drawerLayout = findViewById(R.id.drawer_layout)
        val toggle = ActionBarDrawerToggle(this, drawerLayout, toolBarChats, R.string.nav_drawer_open, R.string.nav_drawer_close)
        toggle.isDrawerIndicatorEnabled
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        nav_view.setNavigationItemSelectedListener(this)
        val header = nav_view.getHeaderView(0)


        reference.addValueEventListener(object : ValueEventListener {
            override fun onCancelled(p0: DatabaseError) {

            }

            override fun onDataChange(p0: DataSnapshot) {
                val user = p0.getValue(User::class.java)

                if (user != null) {
                    navHeader_tvUserName!!.text = user.userName
                    navHeader_tvEmail!!.text = user.geteMail()

                    if (user.imageURL == "default") {
                        navHeader_ivProfile.setImageResource(R.mipmap.ic_launcher_round)
                    } else {
                        Glide
                                .with(this@MainActivity)
                                .load(user.imageURL)
                                .into(navHeader_ivProfile)
                    }
                }

            }
        })
    }


    override fun sendStatus(status: String) {
        mainPresenter.updateStatus(status)
    }


    override fun openStart() {
        startActivity(Intent(applicationContext, StartActivity::class.java))
    }

    override fun openProfile() {
        startActivity(Intent(applicationContext, ProfileActivity::class.java))
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        item.isCheckable = false

        when (item.itemId) {
            R.id.nav_Home -> {
            }
            R.id.nav_LogOut -> {
                mainPresenter.toLogOut()
            }
            R.id.nav_Settings -> {
            }
            R.id.nav_Profile -> {
                mainPresenter.toProfile()
            }
        }
        drawerLayout.closeDrawer(GravityCompat.START)
        return true
    }

    override fun onPostResume() {
        super.onPostResume()
        sendStatus("online")
    }

    override fun onPause() {
        super.onPause()
        sendStatus("offline")
    }

    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            startActivity(Intent(this@MainActivity, StartActivity::class.java))
            finish()
        }
    }
}